docker/build-tools/build-and-push.sh

```diff
@@ -20,8 +20,17 @@ set -eux
 CONTAINER_CLI=${CONTAINER_CLI:-docker}
 
 HUB=${HUB:-gcr.io/istio-testing}
-DATE=$(date +%Y-%m-%dT%H-%M-%S)
 BRANCH=master
+# It is crucially important to use backtick command substitution such that both Docker images
+# created by the build system have the same exact timestamp. Other components of the build system depend
+# on a matching timestamp.
+
+# While backticks are "legacy", they do not operate the same way as $(...). $(...) evaluates
+# each time the variable is evaluated, while backticks evaluate only once and execute
+# a permanent assignment. I think shellcheck has this wrong and the gnu bash docs need a minor update.
```

Comment:
Interesting... How do I replicate this behavior? I tried with https://gist.github.com/jwendell/38ab31ece1394aa7b6e21287dfe2b146 but it works as intended...

Reply:
@jwendell Your test case is invalid. I left comments in the gist as to how to generate the problem.

Cheers
-steve

Reply:
well, I have simplified the the test case a bit:
```
sdake@beast-01:~$ cat jwen.sh 
#!/bin/sh
```
A=$(date +%Y-%m-%dT%H-%M-%S)
echo A=${A}
sleep 2
echo B=${A}
```

And the problem does not reproduce. Let me double-check my assumptions 👍

Reply:
OK, make sure to use /bin/bash instead of sh.